cranelift/wasm/src/code_translator/bounds_checks.rs

```diff
@@ -387,6 +447,9 @@ fn compute_addr(
     // Add the heap base address base
     let base = pos.ins().global_value(addr_ty, heap.base);
 
+    // NB: this addition cannot overflow, since we know that `index <= bound +
+    // guard_size` by the time we reach this code, and correctly configured
+    // heaps cannot have `base + bound + guard_size` overflow.
```

Comment:
I think that this comment isn't actually correct in the case that a spectre guard is used, right? Without spectre guards the previously-trapping condition prevents overflow, but with spectre guards this could overflow. I think the saving grace, though, is that the overflowed value isn't used due to the spectre guard.

Reply:
Yeah I was being a bit loose and fudgy with my wording here, will be more precise.